internal/storage/loopback/loopback.go

```diff
@@ -0,0 +1,191 @@
+package loopback
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	"go.opencensus.io/trace"
+	"golang.org/x/sys/unix"
+
+	"github.com/Microsoft/opengcs/internal/log"
+	"github.com/Microsoft/opengcs/internal/oc"
+	"github.com/pkg/errors"
+)
+
+const (
+	EmptyMode         = 0x0
+	LOOP_CTL_ADD      = 0x4c80
+	LOOP_CTL_GET_FREE = 0x4c82
+	LOOP_CTL_REMOVE   = 0x4c81
+	LOOP_CLR_FD       = 0x4c01
+	LOOP_SET_FD       = 0x4c00
+	SYS_IOCTL         = 16
+)
+
+const ext4Options = "noatime,barrier=0,errors=remount-ro,ro"
+
+// Test dependencies
+var (
+	osMkdirAll  = os.MkdirAll
+	osRemoveAll = os.RemoveAll
+	unixMount   = unix.Mount
+)
+
+// Mount mounts `backingFile` as a loopback device on device number `device`. The loopback device
+// is then ext4 mounted at `target`. If `target` doesn't exist it will be created.
+func Mount(ctx context.Context, device int, target, backingFile string) (err error) {
+	_, span := trace.StartSpan(ctx, "loopback.Mount")
+	defer span.End()
+	defer func() { oc.SetSpanStatus(span, err) }()
+
+	span.AddAttributes(
+		trace.StringAttribute("target", target),
+		trace.Int64Attribute("device", int64(device)),
+		trace.StringAttribute("backingFile", backingFile))
+
+	if err := osMkdirAll(target, 0700); err != nil {
+		return err
+	}
+	defer func() {
+		if err != nil {
+			osRemoveAll(target)
+		}
+	}()
+
+	// If the device already exists `getLoopID` will return -1, so skip this if the device
+	// we're asking for already exists. The UVM starts with 8 loop devices (0,1,2....7)
+	// already so this should be hit exactly 8 times before we have to actually allocate
+	// a new loop device.
+	loopID := device
+	if _, err := os.Stat(fmt.Sprintf("/dev/loop%d", device)); err != nil {
```

Comment:
should we be checking that the error corresponds to no file being present here and then returning the error if not?